runtime/src/accounts_index.rs

```diff
@@ -2989,14 +2904,14 @@ pub mod tests {
         );
 
         assert_eq!(0, account_maps_len_expensive(&index));
-        let w_account_maps = index.get_account_maps_write_lock(&key.pubkey());
-        WriteAccountMapEntry::upsert(
-            w_account_maps,
+        let mut w_account_maps = index.get_account_maps_write_lock(&key.pubkey());
+        w_account_maps.upsert(
             &key.pubkey(),
             new_entry,
             &mut SlotList::default(),
             UPSERT_PREVIOUS_SLOT_ENTRY_WAS_CACHED_FALSE,
         );
+        drop(w_account_maps);
```

Comment:
Can this be collapsed into `index.get_account_maps_write_lock(&key.pubkey()).upsert(...)`? I think that would remove the need for the `drop` too maybe?

Reply:
yes, could be collapsed. just test code and I was trying to change as few lines as possible in these refactorings.